test/userController.test.js

```diff
@@ -94,3 +94,143 @@ describe('GET /api/v1/users/:user_id endpoint', () => {
     expect(response.status).toBe(404);
   });
 });
+
+describe('PATCH /api/v1/users/updateCurrentUser endpoint', () => {
+  it('should update the current user', async () => {
+    const signedCookie = await loginAndReturnCookie({
+      email: 'ava@ava.com',
+      password: 'secret',
+    });
```

Comment:
We can narrow things down to only run this one single test (line 99) with the following Jest command:

`npm run test -- --testPathPattern=test/userController.test.js --testNamePattern="should update the current user"`

*   The first `--` tells npm that we're passing these command options to the underlying `jest` command.
*   `--testPathPattern` lets us give Jest a path as a string or regex that tells it only to run tests in that file (https://jestjs.io/docs/cli#--testpathpatternregex)
*   `--testNamePattern` lets us give Jest a string or regex that it will search for in the file(s) its running the test in; it will only run those tests whose name (either the `decribe` block or the `it` block) matches. We can also shorten this by using `-t="should update the current user"`

Once you do this, you will see that this test fails because the user cannot log in and get a cookie. The login endpoint is returning a `401`. We know that this happens when the credentials that were given to the login endpoint don't match any user in the database. Furthermore, the error log shown in the test output shows that the response did not return any `cookie` so the step where we try to find the `token` in the cookie fails:

```
> bb-practicum-team-4-back@1.0.0 test
> jest --watchAll=false --detectOpenHandles --silent --testPathIgnorePatterns=src "--testPathPattern=test/userController.test.js" "--testNamePattern=should update the current user"

 FAIL  test/userController.test.js
  ● PATCH /api/v1/users/updateCurrentUser endpoint › should update the current user

    TypeError: Cannot read properties of undefined (reading 'find')

      26 |   // Split the cookie string by ';' and take the first part (before the first ';')
      27 |   const signedCookie = cookieHeader
    > 28 |     .find((cookie) => cookie.startsWith('token'))
         |      ^
      29 |     .split(';')[0];
      30 |   return signedCookie; // Return the signed cookie, which typically contains authentication token
      31 | };

      at find (test/userController.test.js:28:6)
      at Object.<anonymous> (test/userController.test.js:109:26)

2023-09-19 09:01:01 [ERROR]: Invalid Credentials - Error: Invalid Credentials
    at login (/Users/aasante/dev/practicum2023/backend/src/controllers/authController.js:69:11)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
2023-09-19 09:01:01 [INFO]: POST /api/v1/auth/login 401 72.801 ms - 29
Test Suites: 1 failed, 1 total
Tests:       1 failed, 9 skipped, 10 total
Snapshots:   0 total
Time:        2.364 s, estimated 4 s
```

The reason this happens, is that each test starts with a clean slate, an empty database. We need to set up ("Arrange") what we want the state of the database to be before our test. 

In this branch, you can see that for example, in the "Get user by id" test on lines 56-63, the test code first "Arranges" the state for the test by creating a test user with the matching email and password. That way, the credentials for logging in will work as expected.

If you pull in the latest changes from `main`, you can see that I've moved the `testUser` creation up into the `beforeAll` hook. This way, each test doesn't need to create that test user and can just go ahead and log in right away as the first "Arrange" step.